checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -170,7 +170,22 @@ protected boolean isInitializationAnnotation(AnnotationMirror anno) {
      * @param field field that might have invariant annotation
      * @return whether or not field has the invariant annotation
      */
-    protected abstract boolean hasFieldInvariantAnnotation(VariableTree field);
+    protected boolean hasFieldInvariantAnnotation(VariableTree field) {
```

Comment:
I would suggest making this method `final`. So far this signature was the extension point and subclasses could implement this method. Now these implementation would stop working, because they should override the newly introduced abstract method below.
By making the method `final` we break such subclasses, which seems nicer than silently failing.
Do you agree?